### PR TITLE
feat(example): Enable scheduler-backed plugin introspection

### PR DESCRIPTION
**Pull request by dcramer**
The example app now wires the scheduler plugin into local chat/dev, loads app plugin modules through the root CLI path, and exposes a `plugins list` jr-rpc capability so the agent can inspect installed plugins. The local dev path now loads the example env consistently, validates database-backed plugins, supports local Postgres services, and keeps local chat QA aligned with the server/plugin wiring.

This also builds out the SQL test foundation needed for that wiring. Junior now has a shared real-Postgres Vitest harness with migrated templates, per-worker databases, rollback fixtures, and CI coverage; product tests can use normal imports because the harness points `JUNIOR_DATABASE_URL` at the worker database. SQL driver selection is now explicit via `JUNIOR_DATABASE_DRIVER`, defaulting to Neon while local Postgres and the harness opt into the `postgres` driver.

The branch also tightens test boundaries by moving heavier coverage out of the unit layer, adding ast-grep lint rules for boundary checks, and documenting the Postgres harness contract. Review the example/CLI plugin path first, then the SQL driver and harness changes, then the test-boundary cleanup.

### CODE REVIEW

File: packages/junior/src/cli/chat.ts [hunk 1]
```diff
@@ -97,6 +109,58 @@ function defaultStateAdapterForLocalChat(): void {
...
+      await localPluginLoader.import<Record<string, unknown>>(pluginModulePath);
+    const pluginSet = mod.plugins ?? mod.default;
+    if (!isPluginSet(pluginSet)) {
+      throw new Error(
+        `${pluginModulePath} must export a defineJuniorPlugins(...) set as "plugins" or default`,
+      );
+    }
+    return pluginSet;
+  }
+  return undefined;
```
[the remark this answers is not in the document]
Comment (dcramer) [reply packages/junior/src/cli/chat.ts]: Leaving this as-is intentionally. Local plugin module discovery should fail fast when a present plugins.js/plugins.mjs/plugins.ts exports an invalid plugin set; skipping to another candidate would mask a stale or broken local plugin configuration and could run Junior with a different plugin catalog than the app author sees on disk.

*— Claude Code*
